spec/lib/vmdb/loggers_spec.rb

```diff
@@ -0,0 +1,52 @@
+describe Vmdb::Loggers do
+  let(:log_file) { Rails.root.join("log", "foo.log").to_s }
+  describe "#create_multicast_logger" do
+    it "defaults the lower level loggers to `DEBUG`" do
+      log = described_class.send(:create_multicast_logger, log_file)
+
+      expect(log.loggers.first.level).to eq(Logger::DEBUG)
+      expect(log.loggers.to_a.size).to   eq(1)
+    end
+
+    context "in a container environment" do
+      around(:each) { |example| in_container_env(example) }
+
+      it "sets logger_instance and $container_log to debug" do
+        log = described_class.send(:create_multicast_logger, log_file)
+
+        expect(log.loggers.first.level).to     eq(Logger::DEBUG)
+        expect(log.loggers.to_a.last.level).to eq(Logger::DEBUG)
+      end
+    end
+  end
+  describe "#apply_config_value" do
+    it "will update the main lower level logger instance" do
+      log = described_class.send(:create_multicast_logger, log_file)
+      described_class.send(:apply_config_value, {:foo => :info}, log, :foo)
+
+      expect(log.loggers.first.level).to eq(Logger::INFO)
+      expect(log.loggers.to_a.size).to   eq(1)
+    end
+
+    context "in a container environment" do
+      around(:each) { |example| in_container_env(example) }
+
+      it "will always keep $container_log as DEBUG" do
+        log = described_class.send(:create_multicast_logger, log_file)
+        described_class.send(:apply_config_value, {:foo => :info}, log, :foo)
+
+        expect(log.loggers.first.level).to     eq(Logger::INFO)
+        expect(log.loggers.to_a.last.level).to eq(Logger::DEBUG)
+      end
+    end
+  end
+end
+
+def in_container_env(example)
```

Comment:
Should this definition be *inside* the describe block?  toplevel clearly infects the global namespace, but I think inside of a describe it will be at least isolated.  @bdunne Do you remember?

Reply:
I can try, since it isn't too much effort to do so.  Wanted to make sure it was not repeated code, but yeah, I can see if tossing it in a describe isolates things.  Couldn't hurt.